.github/workflows/publish-binary.yml

```diff
@@ -0,0 +1,161 @@
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.to }}
+      - name: Setup Rust toolchain
+        run: rustup show
+      - name: Build Node
+        run: cargo build --profile=production --all
+      - name: Save parachain binary
+        if: ${{ matrix.cpu == '' }}
+        run: |
+          mkdir -p binaries
+          cp target/production/tanssi-node binaries/tanssi-node
+          cp target/production/container-chain-template-frontier-node binaries/container-chain-template-frontier-node
+          cp target/production/container-chain-template-simple-node binaries/container-chain-template-simple-node
+      - name: Save parachain custom binary
+        if: ${{ matrix.cpu != '' }}
+        run: |
+          mkdir -p binaries
+          cp target/production/tanssi-node binaries/tanssi-node-${{matrix.cpu}}
+          cp target/production/container-chain-template-frontier-node binaries/container-chain-template-frontier-node-${{matrix.cpu}}
+          cp target/production/container-chain-template-simple-node binaries/container-chain-template-simple-node-${{matrix.cpu}}
+      - name: Upload binaries
+        uses: actions/upload-artifact@v3.1.2
+        with:
+          name: binaries
+          path: binaries
+
+  ####### Prepare the release draft #######
+
+  publish-draft-release:
+    runs-on: ubuntu-latest
+    needs: ["build-binary"]
+    outputs:
+      release_url: ${{ steps.create-release.outputs.html_url }}
+      asset_upload_url: ${{ steps.create-release.outputs.upload_url }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.to }}
+          fetch-depth: 0
+      - uses: actions/download-artifact@v3.0.2
+        with:
+          name: binaries
+          path: build
+      - name: Use Node.js 18.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+      - name: Download Original Tools
+        uses: actions/download-artifact@v3
+        with:
+          name: original-tools
+          path: original-tools
+      - name: Generate release body
+        id: generate-release-body
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        working-directory: original-tools
+        run: |
+          yarn
+          yarn -s run ts-node github/generate-release-body.ts --owner "${{ github.repository_owner }}" --repo "$(basename ${{ github.repository }})" --from ${{ github.event.inputs.from }} --to ${{ github.event.inputs.to }} --srtool-report-folder '../build/' > ../body.md
+      - name: Create draft release
+        id: create-release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.event.inputs.to }}
+          release_name: Tanssi ${{ github.event.inputs.to }}
+          body_path: body.md
+          draft: true
+      - name: Upload tanssi
```

Comment:
we can probably matrix thise part, I didn't try on Moonbeam I think

Reply:
I can try yes